src/test/kotlin/no/skatteetaten/aurora/gobo/integration/boober/VaultServiceTest.kt

```diff
@@ -61,6 +72,66 @@ class VaultServiceTest {
         assertThat(requests.first()?.path).isEqualTo("/v1/vault/$affiliationName")
     }
 
+    @Test
+    fun `Rename vault sunshine`() {
+        val getVaultResponse = BooberVaultBuilder().build()
+        val putVaultResponse = getVaultResponse.copy(name = renamedVaultName)
+        val deleteVaultResponse = getVaultResponse.copy()
+
+        val requests = server.executeBlocking(
+            200 to Response(getVaultResponse),
+            404 to Response(BooberVaultBuilder().build()),
+            200 to Response(putVaultResponse),
+            200 to Response(getVaultResponse),
+            200 to Response(deleteVaultResponse)
+        ) {
+            val renamedVault = vaultService.renameVault(
+                token = "token",
+                RenameVaultInput(
+                    affiliationName = affiliationName,
+                    vaultName = vaultName,
+                    newVaultName = renamedVaultName
+                )
+            )
+            assertThat(renamedVault.name).isEqualTo("renamed-test-vault")
+            assertThat(renamedVault.hasAccess).isTrue()
+            assertThat(renamedVault.permissions).isEqualTo(
+                listOf("APP_PaaS_utv")
+            )
+            assertThat(renamedVault.secrets).isEqualTo(
+                mapOf("latest.properties" to "QVRTX1VTRVJOQU1FPWJtYwp")
+            )
+        }
+    }
+
+    @Test
+    fun `Rename vault failed because new name already exists`() {
+        val getVaultResponse = BooberVaultBuilder().build()
+        val checkIfVaultExistsResponse = BooberVaultBuilder().build()
+        val putVaultResponse = getVaultResponse.copy(name = renamedVaultName)
+        val deleteVaultResponse = getVaultResponse.copy()
+
+        val requests = server.executeBlocking(
+            Response(getVaultResponse),
+            Response(checkIfVaultExistsResponse),
+            putVaultResponse,
+            deleteVaultResponse
+        ) {
+            try {
```

Comment:
her kan du bruke `assertThat(...).isFailure()`

et eksempel her:
```
assertThat {
                probeService.probeFirewall("server.test.no", 9999)
            }.isNotNull().isFailure().all {
                isInstanceOf(WebClientResponseException::class).messageContains("404 Not Found")
            }
```

Reply:
OK, takk for tips :) Endret.